pkg/api/secrets/secrets_test.go

```diff
@@ -0,0 +1,742 @@
+func TestReturnAllSecretsWhenThereAreNoSecrets(t *testing.T) {
+	assert := asserts.New(t)
+	var clients kubernetes.Interface = fake.NewSimpleClientset()
+	mbPairs := map[string]*types.ModelBindingPair{}
+	clusters := []v1beta1.VerrazzanoManagedCluster{}
+	Init(testutilcontroller.NewControllerListers(&clients, clusters, &mbPairs))
+	request, _ := http.NewRequest("GET", "/secrets", nil)
+	recorder := testutil.InvokeHTTPHandler(request, "/secrets", ReturnAllSecrets)
+	assert.Equal(200, recorder.Code)
+	assert.Equal("0", recorder.Header().Get("X-Total-Count"))
+	assert.Equal("[]", strings.TrimSpace(string(recorder.Body.Bytes())))
+}
+
+// func ReturnAllSecrets(w http.ResponseWriter, r *http.Request) {
+func TestReturnAllSecretsWhereModelHasWeblogicSecrets(t *testing.T) {
+	assert := asserts.New(t)
+	var clients kubernetes.Interface = fake.NewSimpleClientset()
+
+	wlsPullSecret := newK8sSecret("test-wls-pull-secret-name-1", constants.DefaultNamespace, "test-wls-pull-secret-id-1", "generic", nil)
+	_, err := clients.CoreV1().Secrets(constants.DefaultNamespace).Create(context.TODO(), &wlsPullSecret, metav1.CreateOptions{})
+	assert.NoError(err)
+
+	wlsConfSecret := newK8sSecret("test-wls-conf-secret-name-1", constants.DefaultNamespace, "test-wls-conf-secret-id-1", "generic", nil)
+	_, err = clients.CoreV1().Secrets(constants.DefaultNamespace).Create(context.TODO(), &wlsConfSecret, metav1.CreateOptions{})
+	assert.NoError(err)
+
+	wlsCredSecret := newK8sSecret("test-wls-cred-secret-name-1", constants.DefaultNamespace, "test-wls-cred-secret-id-1", "generic", nil)
+	_, err = clients.CoreV1().Secrets(constants.DefaultNamespace).Create(context.TODO(), &wlsCredSecret, metav1.CreateOptions{})
+	assert.NoError(err)
+
+	mbPairs := map[string]*types.ModelBindingPair{
+		"test-model-1": {
+			Model: &v1beta1.VerrazzanoModel{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: constants.DefaultNamespace,
+					Name:      "test-model-name-1",
+					UID:       "test-model-uid-1",
+				},
+				Spec: v1beta1.VerrazzanoModelSpec{
+					WeblogicDomains: []v1beta1.VerrazzanoWebLogicDomain{{
+						DomainCRValues: v8.DomainSpec{
+							ConfigOverrideSecrets: []string{wlsConfSecret.Name},
+							ImagePullSecrets: []corev1.LocalObjectReference{{
+								Name: wlsPullSecret.Name,
+							}},
+							WebLogicCredentialsSecret: corev1.SecretReference{
+								Namespace: wlsCredSecret.Namespace,
+								Name:      wlsCredSecret.Name,
+							},
+						},
+					}},
+				},
+			},
+		},
+	}
+	clusters := []v1beta1.VerrazzanoManagedCluster{}
+	Init(testutilcontroller.NewControllerListers(&clients, clusters, &mbPairs))
+	request, _ := http.NewRequest("GET", "/secrets", nil)
+	recorder := testutil.InvokeHTTPHandler(request, "/secrets", ReturnAllSecrets)
+	assert.Equal(200, recorder.Code)
+	ret := make([]Secret, 0)
+	err = json.Unmarshal(recorder.Body.Bytes(), &ret)
+	assert.NoError(err)
+	// Need to invent some way to make the order independent.
```

Comment:
you could iterate over the slice with a `range` statement and have a slice of bools to record when you found them

Reply:
or maybe there is a "contains" assertion available?